frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageHeader.vue

```diff
@@ -100,6 +100,31 @@ export default defineComponent({
     const { user } = usePageUser();
     const { isOwnGroup } = useLoggedInState();
 
+    const canEdit = computed(() => {
+      // Check recipe owner
+      if (!user.id) {
+        return false;
+      }
+      if (user.id === props.recipe.userId) {
+        return true;
+      }
+
+      // Check group and household
+      if (!isOwnGroup.value) {
+        return false;
+      }
+      if (user.householdId !== props.recipe.householdId) {
+        return false;
+      }
+
+      // Check recipe
+      if (props.recipe.settings.locked) {
+        return false;
+      }
+
+      return true;
+    })
```

Comment:
would it be useful to move this into either a function or composable? I'd assume we'd need this in other paces, like dropdown menu's and what not? Moving it into a function would also make it easier to test the logic here.